src/popover.jsx

```diff
@@ -17,30 +17,34 @@ export default class Popover extends React.Component {
     this.state = {
       topOffset: 0,
       leftOffset: 0,
-      isOpen: false
+      isOpen: props.isOpen || false
     };
   }
 
-  componentDidMount() {
-    this.calculateHeights();
+  componentWillReceiveProps(props){
+    if(props.isOpen != this.state.isOpen) this.setState({isOpen: props.isOpen})
+  }
 
-    document.addEventListener('click', (ev) => {
-      ev.stopPropagation();
-      this.handleClick(true, ev);
-    });
+  componentDidMount() {
+    if(this.state.isOpen) this.calculateHeights();
+    if(this.props.closeOnOuterClick !== false){
+      document.addEventListener('click', (ev) => {
+        if(ev.target.dataset['popover']) return ev.stopImmediatePropagation();
+        if(this.state.isOpen == true) this.setState({isOpen: false})
+      });
+    }
+  }
 
-    React.findDOMNode(this.refs.popover).addEventListener('click', (ev) => {
-     if(this.props.stopPropagation === false) return
-     ev.stopPropagation();
-    });
+  componentDidUpdate(prevProps, prevState){
+    if(this.state.isOpen && this.state.isOpen != prevState.isOpen) this.calculateHeights();
+  }
 
-    React.findDOMNode(this.refs.toggleButton).addEventListener('click', (ev) => {
-      ev.stopPropagation();
-      this.handleClick(this.props.isOpen, ev);
-    });
+  toggle = () => {
+    this.setState({isOpen: !this.state.isOpen})
   }
 
   calculateHeights() {
+    if(!this.state.isOpen) return true
```

Comment:
That is bizarre. This happens from time to time (a ref not being there). Sometimes there is a race condition where a component is redrawn and tries to call something via a ref and it just doesn't exist yet. Are you talking inside the aggregator app? If so, maybe this component needs to just recalculate it's dimensions any time it receives new props, then we won't have to call that method from a ref? Seems like that's being seen as an antipattern in the community, and should be something we should try to avoid.